cmd/cluster/etcd_replace.go

```diff
@@ -0,0 +1,183 @@
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type flagOptions struct {
+	nodeId string
+}
+
+// Secrets List
+var secrets = []string{
+	"etcd-peer-",
+	"etcd-serving-",
+	"etcd-serving-metrics-",
+}
+
+// Patch Constants
+const (
+	EtcdQuorumTurnOffPatch = `{"spec": {"unsupportedConfigOverrides": {"useUnsupportedUnsafeNonHANonProductionUnstableEtcd": true}}}`
+	EtcdQuorumTurnOnPatch  = `{"spec": {"unsupportedConfigOverrides": null}}`
+	EtcdForceRedeployPatch = `{"spec": {"forceRedeploymentReason": "single-master-recovery-%s"}}`
+)
+
+func newCmdEtcdMemberReplacement(kubeCli client.Client, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	opts := &flagOptions{}
+	replaceCmd := &cobra.Command{
+		Use:               "etcd-member-replacement",
+		Short:             "Replaces an unhealthy etcd node",
+		Long:              `Replaces an unhealthy ectd node using the member id provided`,
+		Args:              cobra.ExactArgs(0),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(EtcdReplaceMember(kubeCli, *opts, flags))
+		},
+	}
+	replaceCmd.Flags().StringVar(&opts.nodeId, "node", "", "Node ID")
+	return replaceCmd
+}
+
+func EtcdReplaceMember(kubeCli client.Client, opts flagOptions, flags *genericclioptions.ConfigFlags) error {
+	kconfig, clientset, err := getKubeConfigAndClientSet()
+	if err != nil {
+		return err
+	}
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: EtcdLabelSelector,
+	}
+
+	if opts.nodeId == "" {
+		return fmt.Errorf("node name cannot be blank. Please provide node using --node flag")
+	}
+
+	fmt.Println("[INFO] This operation will run with the \"backplane-cluster-admin\" role")
+	// Get the etcd pods
+	pods, err := clientset.CoreV1().Pods(EtcdNamespaceName).List(context.TODO(), listOptions)
+	if err != nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		for _, c := range pod.Status.ContainerStatuses {
+			if c.State.Waiting != nil && (c.State.Waiting.Reason == "CrashLoopBackOff" || c.State.Waiting.Reason == "Error") {
+				podName := "etcd-" + opts.nodeId
+				if podName != pod.ObjectMeta.Name {
+					return fmt.Errorf("the etcd member seems to be healthy. Please run health-check again")
+				}
+				err := ReplaceEtcdMember(kubeCli, kconfig, clientset, flags, opts, pod.ObjectMeta.Name)
+				if err != nil {
+					return err
+				}
+				return nil
+			}
+		}
+	}
+	return fmt.Errorf("none of the etcd members seems to be unhealthy. Please verify once again")
+}
+
+func ReplaceEtcdMember(kubeCli client.Client, kconfig *rest.Config, clientset *kubernetes.Clientset, flags *genericclioptions.ConfigFlags, opts flagOptions, pod string) error {
```

Comment:
Some steps executed as part of this function do not log prior to their execution. It should be very clear to an SRE exactly what is happening throughout this process, so `removeEtcdMember` and `removeEtcdSecrets` should log prior to their call.

This is also important from the perspective of giving SRE context if one of these steps were to error. Currently, the error is passed right back up the chain and dumped out. It should be clear to an SRE to know exactly at what point in the member replacement process that error occurred and what step failed - because the SRE is likely going to have to manually take care of the rest of the steps, and it will help for them to not have to try and guess or figure out where they need to pick up from.